specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json

```diff
@@ -602,7 +859,238 @@
         }
       },
       "description": "The server error."
-    }
+    },
+    "MHSMNetworkRuleSet": {
+        "properties": {
+          "bypass": {
+            "type": "string",
+            "description": "Tells what traffic can bypass network rules. This can be 'AzureServices' or 'None'.  If not specified the default is 'AzureServices'.",
+            "enum": [
+              "AzureServices",
+              "None"
+            ],
+            "x-ms-enum": {
+              "name": "NetworkRuleBypassOptions",
+              "modelAsString": true
```

Comment:
I see all the enums are modelAsString true? Why? Are these meant to be user-extensible? For some SDKs, this can make coding harder / less discoverable. If you expect new values to be returned by older (as this one would be in the future) versions this might also be a good idea.

Reply:
We prefer modelAsString: true because this versions. If the enum is not modeled as a string, then the symbols can NEVER change (if the enum value is persisted in a resource). The reason is because an old client will throw and crash when deserializing a new value that the enum doesn't know about. Modelling as string means no deserialization and the string value just comes back to the old client.

Reply:
Thanks @JeffreyRichter for your answer.